tesseract_task_composer/core/include/tesseract_task_composer/core/task_composer_context.h

```diff
@@ -102,23 +104,41 @@ struct TaskComposerInput
   /** @brief Reset abort and data storage to constructed state */
   void reset();
 
-  bool operator==(const TaskComposerInput& rhs) const;
-  bool operator!=(const TaskComposerInput& rhs) const;
+  /**
+   * @brief Create a child context which is used for dynamic tasking
+   * @details
+   *   - Everything is shared between the parent and the child except abort_ and task_infos_.
+   *   - Task infos should be merged into the parent by the developer
+   */
+  TaskComposerContext::UPtr createChild();
```

Comment:
Seems like this is basically just a copy constructor and should utilize that function instead of this new one

Reply:
Not exactly, I would expect a copy constructor and copy assignment to copy all member variables which this method does not. Not sure what the best practices are this, but will look.

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c61-a-copy-operation-should-copy

Reply:
Method has been removed because it is no longer needed.